specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/instanceFailoverGroups.json

```diff
@@ -0,0 +1,644 @@
+          "type": "integer"
+        }
+      }
+    },
+    "InstanceFailoverGroupReadOnlyEndpoint": {
+      "description": "Read-only endpoint of the failover group instance.",
+      "type": "object",
+      "properties": {
+        "failoverPolicy": {
+          "description": "Failover policy of the read-only endpoint for the failover group.",
+          "enum": [
+            "Disabled",
+            "Enabled"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ReadOnlyEndpointFailoverPolicy",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "PartnerRegionInfo": {
+      "description": "Partner region information for the failover group.",
+      "type": "object",
+      "properties": {
+        "location": {
+          "description": "Geo location of the partner managed instances.",
+          "type": "string"
+        },
+        "replicationRole": {
+          "description": "Replication role of the partner managed instances.",
+          "enum": [
+            "Primary",
+            "Secondary"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "InstanceFailoverGroupReplicationRole",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "ManagedInstancePairInfo": {
+      "description": "Pairs of Managed Instances in the failover group.",
+      "type": "object",
+      "properties": {
+        "primaryManagedInstanceId": {
+          "description": "Id of Primary Managed Instance in pair.",
+          "type": "string"
+        },
+        "partnerManagedInstanceId": {
+          "description": "Id of Partner Managed Instance in pair.",
+          "type": "string"
+        }
+      }
+    },
+    "Resource": {
+      "description": "ARM resource.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Resource ID.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "Resource type.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "ProxyResource": {
+      "description": "ARM proxy resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {}
+    },
+    "InstanceFailoverGroup": {
+      "description": "An instance failover group.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
```

Comment:
Does it need to have properties within properties?

Reply:
Isn't this right? Outer "properties" is swagger syntax, inner "properties" is the name of the actual json element.